include/IECoreScene/Primitive.inl

```diff
@@ -38,6 +38,67 @@
 namespace IECoreScene
 {
 
+template<typename T> boost::optional<PrimitiveVariable::IndexedView < typename T::ValueType::value_type> >
+Primitive::variableIndexedView( const std::string &name, PrimitiveVariable::Interpolation requiredInterpolation, bool throwIfInvalid ) const
+{
+	PrimitiveVariableMap::const_iterator it = variables.find( name );
+	if( it == variables.end() )
+	{
+		if( throwIfInvalid )
+		{
+			throw IECore::InvalidArgumentException( boost::str( boost::format( "Primitive::variableIndexedView - No primvar named '%1%' found" ) % name ) );
+		}
+		else
+		{
+			return boost::none;
+		}
+	}
+
+	if( it->second.interpolation != requiredInterpolation )
```

Comment:
To match `variableData()`, an unspecified interpolation should mean that any interpolation is accepted :
```
	if( requiredInterpolation!=PrimitiveVariable::Invalid && it->second.interpolation!=requiredInterpolation )
```

Sorry, I should have caught this at the same time as the original logic error I pointed out...